Parsers/SymantecDLP/SymantecDLP.txt

```diff
@@ -0,0 +1,51 @@
+// Title:           Symantec Data Loss Prevention (DLP) Parser
+// Author:          Microsoft
+// Version:         1.0
+// Last Updated:    11/24/2020
+// Comment:         Initial Release
+//  
+// DESCRIPTION:
+// This parser takes raw Symantec DLP logs from a Syslog stream and parses the logs into a normalized schema.
+//
+// USAGE:
+// 1. Open Log Analytics/Azure Sentinel Logs blade. Copy the query below and paste into the Logs query window. 
+// 2. Click the Save button above the query. A pane will appear on the right, select "as Function" from the drop down. Enter a Function Name.
+//    It is recommended to name the Function Alias, as SymantecDLP
+// 3. Kusto Functions can typically take up to 15 minutes to activate. You can then use Function Alias for other queries.
+//
+// REFERENCES: 
+// Using functions in Azure monitor log queries: https://docs.microsoft.com/azure/azure-monitor/log-query/functions
+// 
+// LOG SAMPLES:
+// This parser assumes the raw log are formatted as follows:
+//
+//<13>Sep 5 08:22:01 data.example.com CEF:0|Symantec|DLP|12.5.0|ruleID|Policy|5|BLOCKED=Passed INCIDENT_ID=204529 INCIDENT_SNAPSHOT=https://main.example.com/Path/Address MATCH_COUNT=3 PROTOCOL=SMTP RECIPIENTS=email@example.com SENDER=sender@example.com SUBJECT=Sensitive Data (attachment included) SEVERITY=1:High FILE_NAME=myfile.xml
+//
+// 
+CommonSecurityLog
+| where DeviceVendor == "Symantec" and DeviceProduct == "DLP"
+| extend Parser = extract_all(@"(^|;)(?P<key>[^=]+)\=(?P<value>[^;]+)(;|$)", dynamic(["key","value"]), AdditionalExtensions)
```

Comment:
I don't think this is working as expected on the AdditionalExtensions field.  When I simply use the sample entry you have above and put that in my own table with a column named AdditionalExtensions, it doesn't appear to parse it properly.

![image](https://user-images.githubusercontent.com/45466083/100818836-c9c8f700-33ff-11eb-9080-725693832c80.png)

If we then do the mv-apply and projecting of columns, they are all empty -

![image](https://user-images.githubusercontent.com/45466083/100818929-fb41c280-33ff-11eb-90fe-8c886916d5d2.png)

Reply:
@shainw - I uploaded a new parser, reverting back to the old approach.